shopfloor/services/zone_picking.py

```diff
@@ -233,15 +235,46 @@ def _data_for_location(self, zone_location, picking_type, location):
             "location": self.data.location(location),
         }
 
+    def _zone_lines(self, zones):
+        return self.env["stock.move.line"].search(
+            self._find_location_move_lines_domain(zones)
+        )
+
+    def _data_for_select_zone(self, zones):
+        """Retrieve detailed info for each zone.
+
+        Zone without lines are skipped.
+        Zone with lines will have line counters by operation type.
+
+        :param zones: zone location recordset
+        :return: see _schema_for_select_zone
+        """
+        res = []
+        for zone in zones:
+            zone_data = self.data.location(zone)
+            zone_lines = self._zone_lines(zone)
+            if not zone_lines:
+                continue
+            zone_data["operation_types"] = []
+            for picking_type, lines in groupby(
```

Comment:
`groupby` must be called on lines sorted by picking type, are we sure they are?

Reply:
I don't get it, sorry. We group by picking type here and we only want to count lines by picking type. What is the point of sorting them?

Reply:
It's a requirement of `groupby` https://docs.python.org/3/library/itertools.html#itertools.groupby.


```
    # [k for k, g in groupby('AAAABBBCCDAABBB')] --> A B C D A B
    # [list(g) for k, g in groupby('AAAABBBCCD')] --> AAAA BBB CC D
```

Reply:
damn, you are right, I completely flushed this notion from my mind :sweat:

Reply:
should be better now, tnx for catching it!